test/functional/lua/net_spec.lua

```diff
@@ -0,0 +1,60 @@
+local n = require('test.functional.testnvim')()
+local t = require('test.testutil')
+local skip_integ = os.getenv('NVIM_TEST_INTEG') ~= '1'
+
+local exec_lua = n.exec_lua
+
+local function assert_404_error(err)
+  assert(
+    err:lower():find('404') or err:find('22'),
+    'Expected HTTP 404 or exit code 22, got: ' .. tostring(err)
+  )
+end
+
+describe('vim.net.request', function()
+  before_each(function()
+    n:clear()
+  end)
+
+  it('fetches a URL into memory (async success)', function()
```

Comment:
let's also have a test that tests text response and binary response (jpg or something)